### PR TITLE
feat(codefix): add parse utils for template only components

### PR DESCRIPTION
**Pull request by wondersloth**
- `getNearestTemplateOnlyComponentVariableDeclaration` will be used to walk up from a glint args diagnostic to a template only declaration.
- `getComponentSignatureNameFromTemplateOnlyComponent` will be used to grab an existing signature for a TOC.
- Supports feature work for  #1227 
- Refactor getNearest methods to use `ts.findAncestor` library method

### CODE REVIEW

File: packages/codefixes/src/fixes/glint/glint-parsing-utils.ts [hunk 1]
```diff
@@ -243,3 +243,57 @@ export function getJSDocExtendsTagWithSignature(
...
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+export function getNearestTemplateOnlyComponentVariableDeclaration(
+  targetNode: ts.Node,
+  identifierName = 'TemplateOnlyComponent'
```
Comment (egorio) [inline packages/codefixes/src/fixes/glint/glint-parsing-utils.ts:270]: Can this value be a constant?

**Review by egorio — APPROVED** [review]

File: packages/codefixes/src/fixes/glint/glint-parsing-utils.ts [hunk 2]
```diff
@@ -243,3 +243,57 @@ export function getJSDocExtendsTagWithSignature(
...
+  let target = targetNode;
+  do {
+    if (!target.parent) {
+      return;
+    }
+
+    if (isTemplateOnlyComponent(target, identifierName)) {
+      return target;
+    }
+  } while ((target = target.parent));
```
Comment (egorio) [inline packages/codefixes/src/fixes/glint/glint-parsing-utils.ts:281]: You might want to try ts.findAncestor, like this: https://github.com/rehearsal-js/rehearsal-js/blob/master/packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance.ts#L65

Comment (wondersloth) [reply packages/codefixes/src/fixes/glint/glint-parsing-utils.ts:281]: Thank you for the recommendation. Will update this to use the library method.

Comment (wondersloth) [reply packages/codefixes/src/fixes/glint/glint-parsing-utils.ts]: I've pushed up  changes to the two methods that were finding up.

**Review by egorio — APPROVED** [review]

**Pull request merged**